src/libraries/System.Runtime.InteropServices/gen/ComInterfaceGenerator/ComInterfaceGenerator.cs

```diff
@@ -541,7 +541,7 @@ private static InterfaceDeclarationSyntax GenerateImplementationVTable(ComInterf
                                 BinaryExpression(
                                     SyntaxKind.MultiplyExpression,
                                     SizeOfExpression(TypeSyntaxes.VoidStar),
-                                    IntLiteral(3 + interfaceMethods.Methods.Length))))));
+                                    IntLiteral(1 + interfaceMethods.Methods.Max(m => m.GenerationContext.VtableIndexData.Index)))))));
```

Comment:
@jtschuster based on my testing, there are situations where interfaceMethods.Methods.Length == 0 and in this case Enumerable.Max() throws an exception.
```suggestion
                                    IntLiteral(1 + interfaceMethods.Methods.DefaultIfEmpty(0).Max(m => m.GenerationContext.VtableIndexData.Index)))))));
```